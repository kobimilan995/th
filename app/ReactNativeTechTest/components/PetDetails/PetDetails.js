import React from 'react';
import {View, Text, StyleSheet} from 'react-native';
import PropTypes from 'prop-types';

const PetDetails = ({pet}) => {
  return (
    <View style={styles.container}>
      <View style={styles.iconContainer} />
      <Text style={styles.petNameText}>{pet.name}</Text>
      <Text style={styles.petTypeText}>{pet.type}</Text>
      <Text>
        Needs feeding{' '}
        {pet.feeds === 1 ? 'once a day.' : `${pet.feeds} times a day.`}
      </Text>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    justifyContent: 'center',
    alignItems: 'center',
  },
  iconContainer: {
    height: 60,
    width: 60,
    borderRadius: 45,
    backgroundColor: 'grey',
    marginTop: -65,
    marginBottom: 30,
  },

  petNameText: {
    fontSize: 24,
    fontWeight: 'bold',
    marginBottom: 20,
  },

  petTypeText: {
    fontSize: 18,
    marginBottom: 20,
  },
});

PetDetails.propTypes = {
  pet: PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired,
    feeds: PropTypes.number.isRequired,
  }).isRequired,
};

export default PetDetails;
