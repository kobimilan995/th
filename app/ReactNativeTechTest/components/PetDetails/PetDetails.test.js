/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';

import PetDetails from './PetDetails';

describe('Components - PetDetails', () => {
  it('renders - light mode', () => {
    const tree = renderer
      .create(
        <PetDetails
          pet={{
            name: 'Bowser',
            type: 'Rottweiler',
            feeds: 4,
            id: 1,
          }}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
