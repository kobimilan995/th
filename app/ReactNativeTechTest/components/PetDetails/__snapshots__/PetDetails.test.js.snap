// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components - PetDetails renders - light mode 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "grey",
        "borderRadius": 45,
        "height": 60,
        "marginBottom": 30,
        "marginTop": -65,
        "width": 60,
      }
    }
  />
  <Text
    style={
      Object {
        "fontSize": 24,
        "fontWeight": "bold",
        "marginBottom": 20,
      }
    }
  >
    Bowser
  </Text>
  <Text
    style={
      Object {
        "fontSize": 18,
        "marginBottom": 20,
      }
    }
  >
    Rottweiler
  </Text>
  <Text>
    Needs feeding
     
    4 times a day.
  </Text>
</View>
`;
